config/config.go

```diff
@@ -165,15 +177,25 @@ func GetDefaultConfigDir() (dir string, err error) {
 	return dir, nil
 }
 
-func expandPath(s string) string {
-	if len(s) >= 2 && s[0] == '~' && os.IsPathSeparator(s[1]) {
-		if runtime.GOOS == "windows" {
-			s = filepath.Join(os.Getenv("USERPROFILE"), s[2:])
-		} else {
-			s = filepath.Join(os.Getenv("HOME"), s[2:])
+// Given a path to either a file or directory, returns its absolute path format.
+// ExpandPath resolves "~/" prefix in a given system path.
+// Raise error if path is an empty string as it
+func ExpandPath(path string) (string, error) {
+	if path == "" {
+		error := errors.New("path to file/directory is not set.")
+		return path, error
+	}
+
+	if strings.HasPrefix(path, "~/") {
```

Comment:
How would this be on Windows? Doesn't that use backslashes?

Maybe we should just check for tilde prefix instead and no slash, wdyt?

Reply:
fair point. 

perhaps keeping the original conditions wouldn't be too bad:
`if s[0] == '~' && os.IsPathSeparator(s[1])`

Reply:
I would even take this a bit further: turn it into a small function with a nice name. 

IsRelativePath(string)

Then we can write some tests for it and make sure it works as expected, and it can be re-used elsewhere too. But mainly, it is more readable. When I read `isRelativePath(path)` i directly understand what's going on.

vs reading `if s[0] == '~' && os.IsPathSeparator(s[1])` needs at least a tiny bit of thinking more.

less thinking == better code